benchmarks/nnls/objective.py

```diff
@@ -0,0 +1,30 @@
+from benchopt.base import BaseObjective
+import numpy as np
+
+
+class Objective(BaseObjective):
+    name = "Non Negative Least Squares"
+
+    parameters = {
+        'fit_intercept': [False],
+    }
+
+    def __init__(self, fit_intercept=False):
+        self.fit_intercept = fit_intercept
+
+        super().__init__()
+
+    def set_data(self, X, y):
+        self.X, self.y = X, y
+
+    def __call__(self, beta):
+        # if (beta > 0).all():
+        #     diff = self.y - self.X.dot(beta)
+        #     return .5 * diff.dot(diff)
+        # else:
+        #     return np.inf
```

Comment:
Why not checking feasibility of solution ?

Reply:
it was debugging mode for a while and then I forgot...
will add it back.

Reply:
nitpick but the testing should be inverted: (beta < 0).any() is faster to evaluate

Reply:
yes, I was not clear, if you use (beta < 0).any() the action associated is return np.inf